elastic/elastic-agent
# [windows] if `elastic-agent run` fails, log error to Application EventLog

## What does this PR do?

On Windows if the run command fails, then the error message will be sent to the Application EventLog.  Other OSes do not require this since they capture stderr from processes that run as services, Windows does not.

This is only for the `run` command because the other commands should be run from the CLI, where the error is already published on stderr.

## Why is it important?

If any error happens before the elastic-agent internal logging is setup and the run command exits while running as a Windows service, the error is lost.  Without this error it is very difficult to determine why `elastic-agent run` is failing.

## Checklist


## Disruptive User Impact


## How to test this PR locally

1. Install `elastic-agent` with `-a` flag pointing to CA certificates on the local filesystem
2. Stop `elastic-agent` service
3. Rename the CA certificates
4. Try to start `elastic-agent` service
5. Verify that Application EventLog has event showing that elastic-agent failed and what the error was.

## Related issues

- Closes #4627

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

## Screenshots
<img width="1706" alt="Screenshot 2024-06-04 at 10 19 55" src="https://github.com/elastic/elastic-agent/assets/57081003/8c6ced9f-927f-4143-9182-83765d5d4ea2">
